config-store-yaml/src/main/java/io/stargate/config/store/api/yaml/ConfigStoreActivator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store.api.yaml;
+
+import io.stargate.config.store.api.ConfigStore;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Hashtable;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigStoreActivator implements BundleActivator {
+  private static final Logger logger = LoggerFactory.getLogger(ConfigStoreActivator.class);
+  private final String configYamlLocation;
+
+  // for testing purpose
+  public ConfigStoreActivator(String configYamlLocation) {
+    this.configYamlLocation = configYamlLocation;
+  }
+
+  public ConfigStoreActivator() {
+    this(
+        System.getProperty(
+            "stargate.config_store.yaml.location", "/etc/stargate/stargate-config.yaml"));
+  }
+
+  @Override
+  public void start(BundleContext context) {
+    logger.info("Starting Config Store YAML...");
+    Path yamlFilePath = Paths.get(configYamlLocation);
+    if (!Files.exists(yamlFilePath)) {
+      logger.error(
+          "The yaml file does not exists, please check the path: {}. The ConfigStoreYaml will not be registered.",
+          yamlFilePath);
+      return;
```

Comment:
Shouldn't we throw an exception and fail startup? Otherwise this is just going to cause stargate init to hang, right?

Reply:
Why would it block? I've tested and it does not - it just not registering the extension. 
I think that we can change it to throw exception depending on how the implementation here:
https://github.com/riptano/cloud-ondemand/pull/2339/files#diff-15daad6661ac657b68d9b5ddf39c9e77bd60f7840b447a264e22fc98f9c9c648R9
will look like

Reply:
I guess Doug meant that if some other module depends on the config service, it will never really register in this case too effectively making those other services/APIs unavailable.

I think it would be preferable to throw an exception is this case and fail the whole Stargate startup process for a clear and unambiguous failure mode.

Reply:
Yes, I also think that being explicit is a good choice but there are two problems with throwing here:
1. If we will merge it and throw the exception, then people will not be able to start the stargate any more without providing this new `stargate-config.yaml` file. I think that it may be too intrusive.
2. We need to coordinate with the cloud-on-demand changes (https://github.com/riptano/cloud-ondemand/pull/2339) - only after our deployment process will be adapted, we can throw an exception.

So my proposition is to log an error for now to allow easier transition and change this logic to throw an exception once the deployment process will be adapted. wdyt?

Reply:
Why not add an empty default config yaml file? This way we will defer errors to specific modules that need config settings while allowing ordinary modules to activate normally?

Reply:
We will have one config file provided by the kubernetes; see this:
https://github.com/riptano/cloud-ondemand/pull/2339
If we provide the default file here, there will be a conflict with the file provided by the k8s.
We may decide to add a default file or file per module once we will decide about overrides; for more info, please see this discussion:
https://github.com/stargate/stargate/issues/302#issuecomment-716535829
but also this should be done separately.

Reply:
my concern is that k8s does not apply to local builds and to our public docker images... those will effectively not have a config service unless someone adds a config file manually.... I'm not 100% sure what the best solution could be in this case, but in my previous experience with OSGi debugging missing service instances is very tricky and tedious... dependent services just wait without any visible error messages.

Reply:
We can add a temporary feature flag for this with a system prop that defaults to disabled. But like Dmitri I've had enough fun debugging OSGI, we should fail fast if at all possible